### PR TITLE
HSEARCH-3246 Generate Coveralls and Sonar reports from the Jenkinsfile instead of Travis

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3246

~Based on #1723 , which should be merged first.~ => Done

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -320,6 +327,24 @@ stage('Default build') {
...
+									-DpullRequest=${env.CHANGE_ID} \\
+							""" : """ \\
+									-Dbranch=${env.BRANCH_NAME} \\
+							"""} \\
+					"""
+				}
+			}
+			catch (CredentialNotFoundException e) {
+				echo "No Coveralls token configured - skipping Coveralls report. Error was: ${e}"
+			}
```
Comment (gsmet) [inline Jenkinsfile]: You prefer skipping it (and probably it going unnoticed) rather than throwing the exception?

File: Jenkinsfile [hunk 2]
```diff
@@ -61,6 +49,43 @@ import org.jenkinsci.plugins.credentialsbinding.impl.CredentialNotFoundException
...
+ *       email:
+ *         recipients: ... # string containing a space-separated list of email addresses to notify in case of failing non-PR builds>
+ *
+ * #### Credentials
+ *
+ * The following credentials are necessary for some features:
+ *
+ * - 'aws-elasticsearch' AWS credentials, to test Elasticsearch as a service on AWS
+ * - 'coveralls-repository-token' secret text credentials containing the repository token,
+ * to send coverage reports to coveralls.io. Note these credentials should be registered at the job level, not system-wide.
```
Comment (gsmet) [inline Jenkinsfile]: Does this remark apply to all three credentials or only this one?

File: Jenkinsfile (hunk 1 above)
Comment (yrodiere) [reply Jenkinsfile]: It's for the benefit of anyone configuring a job for his fork and not wanting to also set up coveralls, sonar, and so on. For example if you (@gsmet) set up a job for your own fork, I suppose you'll want the job to work out of the box, and only later, maybe, set up additional integrations.

Note that we only ignore the `CredentialNotFoundException`. As long as you provide credentials, errors should be reported correctly and fail the build. Which means in particular that, for the job currently configured on master, this `catch` block has no effect.

File: Jenkinsfile (hunk 2 above)
Comment (yrodiere) [reply Jenkinsfile]: Only this one. The token is used both for authentication of the user and of the repository, so we can't use a global one. Other credentials are only used to authenticate the user.

**Comment by yrodiere** [comment]
Merging, thanks

**Pull request merged**